React_Notes_Block/src/pages/mainNotes/listNotes/ListNotes.jsx

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-unused-expressions */
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Typography, Grid, Box } from '@material-ui/core';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import Stack from '@mui/material/Stack';
+
+import useStyles from './styled';
+
+import WAHA_NOTES from '../../../config/constants/initNoteData';
+import chooseNote from '../../../utils/ChooseNote';
+import callToEditNote from '../../../utils/CallToEditNote';
+import saveEditedNote from '../../../utils/SaveEditedNote';
+import sliceDescription from '../../../utils/SliceDescription';
+import deleteNote from '../../../utils/DeleteNote';
+import ButtonEdit from '../../../components/notes/actionsButtons/ButtonEdit';
+import InputChange from '../../../components/notes/InputChange';
+import setToLocalStorage, {
+  getToLocalStorage,
+} from '../../../utils/localStorage/SetGetLocStor';
+
+const ListNotes = ({ sentAbout }) => {
+  const [noteList, setNoteList] = useState(WAHA_NOTES);
+  const {
+    gridNotes,
+    listItemActive,
+    listItem,
+    noteText,
+    title,
+    description,
+    noteActions,
+  } = useStyles();
+
+  const handleItem = (id, currentState = noteList, actionFunction) => {
+    const filtredNotes = actionFunction(id, currentState);
+    setNoteList(filtredNotes);
+  };
+  const handleDelete = (id, currentState = noteList) => {
+    const actualNotes = deleteNote(id, currentState);
+    setNoteList(actualNotes);
+  };
+  const handleSaveNote = (id, valueFromInput) => {
+    const savedNotes = saveEditedNote(id, noteList, valueFromInput);
+    setNoteList(savedNotes);
+  };
+
+  useEffect(() => {
+    const localNote = getToLocalStorage();
+    if (localNote) {
+      setNoteList(localNote);
+    }
+  }, []);
+  useEffect(() => {
+    sentAbout(noteList);
+    !noteList.length
+      ? setToLocalStorage(WAHA_NOTES)
+      : setToLocalStorage(noteList);
+  }, [noteList]);
+  return (
+    <Grid container spacing={2} className={gridNotes}>
+      {noteList
+        ? noteList.map((item, index) => (
```

Comment:
We can use here optional chaining.
```suggestion
      {noteList?.map((item, index) => (
```

Reply:
Or `noteList && noteList.map..`
It's not a good practice to use `null` inside jsx

Reply:
```suggestion
      {noteList
        ? noteList.map((item, index) => (
```
Remake notelist and added other component instead null